### PR TITLE
start refactoring Minigames

### PR DESCRIPTION
**Pull request by maxbachmann**

### CODE REVIEW

File: PublishedModules/Psychokiller1888/Minigames/Minigames.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def __init__(self):
 				klass = getattr(lib, game)
 				minigame = klass()
 				self._minigames[game] = minigame
-				self._SUPPORTED_INTENTS += minigame.intents
+				self._INTENTS += minigame.intents
```
Comment (maxbachmann) [inline PublishedModules/Psychokiller1888/Minigames/Minigames.py]: I understand that each of the games can add own intents. However I have no clue how the Module class knows about them

Comment (maxbachmann) [reply PublishedModules/Psychokiller1888/Minigames/Minigames.py]: well they could be added by converting the list to a dict that calls the onMessage method of the current game

File: PublishedModules/Psychokiller1888/Minigames/Minigames.py [hunk 2]
```diff
@@ -50,8 +50,7 @@ def __init__(self):
 				lib = importlib.import_module(f'modules.Minigames.model.{game}')
 				klass = getattr(lib, game)
 				minigame = klass()
-				self._minigames[game] = minigame
-				self._SUPPORTED_INTENTS += minigame.intents
+				self._INTENTS = {**self._INTENTS, **dict.fromkeys(minigame.intents, self.minigameIntent)}
```
Comment (maxbachmann) [inline PublishedModules/Psychokiller1888/Minigames/Minigames.py:53]: how does alice know about these intents, when they are added to the variable after it got passed to the parent

File: PublishedModules/Psychokiller1888/Minigames/Minigames.py [hunk 3]
```diff
@@ -65,65 +64,77 @@ def onUserCancel(self, session: DialogSession):
...
-						text=self.randomTalk('endPlaying')
-					)
 			else:
-				self._minigame.onMessage(intent, session)
+				game = session.slotValue('WhichGame')
+				self._minigame = self._minigames[game]
+				self._minigame.start(session)
 
+		elif self._minigame is not None:
+			self._minigame.onMessage(intent, session)
```
Comment (maxbachmann) [inline PublishedModules/Psychokiller1888/Minigames/Minigames.py:98]: can this intent even occur while a game is running? as far as I did see it should always use a intent filter that does not allow this

Comment (maxbachmann) [reply PublishedModules/Psychokiller1888/Minigames/Minigames.py:98]: urghh kind of ugly that they show the complete diff in the review

**Comment by maxbachmann** [comment]
just tested and definetly broken.
Did test the old version aswell. It worked better, but when playing rock paper scissors it did not pass the intent to the game either

**Comment by Psychokiller1888** [comment]
Don't merge to master, but to alpha 3 :) Will take a look at them

**Comment by maxbachmann** [comment]
ok merged. Did definetly screw up some stuff ^^

**Comment by Psychokiller1888** [comment]
Ok, will look, but I need a state machine first. Started the work on it, but realized there's more to it. It's not a state machine we need, more like a dynamic mapping base on 3 conditions, the intent, the previous intent and a state. Because in one module, a YesOrNo intent can be used by different parts, with same previous intents but not same states. The whole thing is to make something understandable for the end user

**Pull request closed without merging**